Tools/AP_Bootloader/support.cpp

```diff
@@ -538,11 +538,23 @@ void port_setbaud(uint32_t baudrate)
   check if flash has any ECC errors and if it does then erase all of
   flash
  */
+#define ECC_CHECK_CHUNK_SIZE (32*sizeof(uint32_t))
 void check_ecc_errors(void)
 {
     __disable_fault_irq();
+    // stm32_flash_corrupt(0x8043200);
     auto *dma = dmaStreamAlloc(STM32_DMA_STREAM_ID(1, 1), 0, nullptr, nullptr);
+
+#if HAL_DMA_SAFE_MEM_START
+    uint32_t *buf = (uint32_t *)HAL_DMA_SAFE_MEM_START;
```

Comment:
need to check that check_err_errors() is called before we do heap initialisation, or if we do heap init then we should call malloc_dma() (ie. check if CH_CFG_USE_HEAP == TRUE then use malloc_dma())

Reply:
simpler is to always enable CH_CFG_USE_HEAP on H7, then unconditionally use malloc_dma(), that gives very simple logic

Reply:
modified to use CH_CFG_USE_HEAP if Flash size for bootloader >128K